File: testdata/SampleSolution/SampleApp/Data/EfCoreStubs.cs
// Minimal EF Core stubs for SampleSolution.
// SampleApp is a classlib that does not reference the real Microsoft.EntityFrameworkCore package.
// These stubs allow AppDbContext.cs to compile without a NuGet dependency.
// The DbTableExtractor uses name-based type detection and works with these stubs.
namespace Microsoft.EntityFrameworkCore;

public class DatabaseFacade
{
    public int ExecuteSqlRaw(string sql, params object[] parameters) => 0;
    public int ExecuteSqlInterpolated(FormattableString sql) => 0;
}

public class DbContext
{
    public DatabaseFacade Database { get; } = new DatabaseFacade();
}

public class DbSet<T> where T : class { }

File: testdata/SampleVbSolution/SampleVbApp/Data/EfCoreStubs.vb
Namespace Stubs

    ''' <summary>Minimal EF Core stubs so classlib compiles without real EF NuGet.</summary>
    Public Class DbContext
        Public Overridable Sub OnModelCreating(builder As Object)
        End Sub
    End Class

    Public Class DbSet(Of T)
    End Class

End Namespace
